src/bot/commands/misc/convert.js

```diff
@@ -43,6 +53,12 @@ function convertUnits (baseValue, baseType, targetType) {
  * @returns {Promise<string, Error>} Message to the user if successful, rejects generic error if unsuccessful
  */
 async function convertCurrency (baseValue, baseType, targetType) {
+	const acceptedUnits = Object.values(unitAliases);
+	const isValidUnits = acceptedUnits.includes(baseType) && acceptedUnits.includes(targetType);
+	if (!isValidUnits) {
+		throw new Error('Incorrect Currency Unit');
+	}
+
```

Comment:
`unitAliases` doesn't have to do with currencies. I think you meant to check against the keys of `rates` instead?

It should be enough to just check if `rates[baseType]` and `rates[targetType]` are defined (i.e. `!== undefined` or `!= null`), rather than pulling all the entries into a new array and doing a scan.

Also, we should probably move this check down past the lines where we uppercase the currency types, since we know keys in that object should always be uppercase.

Reply:
sorry, I thought the alias values ​​were the types of currencies you work with, I'll fix it